syntax = "proto3";

package echo;

option go_package = "grpc-example/gen/echopb";

// EchoService demonstrates bidirectional gRPC communication
service EchoService {
  // Echo returns the message back with responder info
  rpc Echo(EchoRequest) returns (EchoResponse);

  // EchoStream enables bidirectional streaming
  rpc EchoStream(stream EchoRequest) returns (stream EchoResponse);
}

message EchoRequest {
  string message = 1;
  string sender = 2;
}

message EchoResponse {
  string message = 1;
  string responder = 2;
  int64 timestamp = 3;
}
